packages/zoe/src/contracts/autoswap.js

```diff
@@ -23,245 +22,233 @@ import '../../exported';
  * adding and removing liquidity, and for doing a swap. Other API operations
  * support monitoring the price and the size of the liquidity pool.
  *
- * @param {ContractFacet} zcf
+ * @type {ContractStartFn}
  */
-const makeContract = zcf => {
-  // Create the liquidity mint and issuer.
+const start = async (zcf, _terms) => {
+  // Create a local liquidity mint and issuer.
+  const liquidityMint = await zcf.makeZCFMint('Liquidity');
+  // AWAIT  ////////////////////
+
   const {
-    mint: liquidityMint,
     issuer: liquidityIssuer,
     amountMath: liquidityAmountMath,
-  } = makeIssuerKit('liquidity');
-
+  } = liquidityMint.getIssuerRecord();
   let liqTokenSupply = 0;
 
-  const {
-    makeEmptyOffer,
-    checkHook,
-    escrowAndAllocateTo,
-    assertNatMathHelpers,
-    trade,
-  } = makeZoeHelpers(zcf);
-
-  return zcf.addNewIssuer(liquidityIssuer, 'Liquidity').then(() => {
-    const { brandKeywordRecord } = zcf.getInstanceRecord();
-    Object.values(brandKeywordRecord).forEach(brand =>
-      assertNatMathHelpers(brand),
-    );
-    const getPoolKeyword = brandToMatch => {
-      const entries = Object.entries(brandKeywordRecord);
-      for (const [keyword, brand] of entries) {
-        if (brand === brandToMatch) {
-          return keyword;
-        }
+  const { brandKeywordRecord } = zcf.getInstanceRecord();
+  Object.values(brandKeywordRecord).forEach(brand =>
+    assertNatMathHelpers(zcf, brand),
+  );
+  const getPoolKeyword = brandToMatch => {
+    const entries = Object.entries(brandKeywordRecord);
+    for (const [keyword, brand] of entries) {
+      if (brand === brandToMatch) {
+        return keyword;
       }
-      throw new Error('getPoolKeyword: brand not found');
-    };
+    }
+    throw new Error('getPoolKeyword: brand not found');
+  };
+
+  const { zcfSeat: poolSeat } = await zcf.addEmptySeat();
```

Comment:
is this meant to be async? If so, why not use makeEmptyOffer?

Reply:
Dropped the `async`. Thanks.